### ISSUE TITLE
HELP!! When there's an update tip on the top, I check branch and lost all of my uncommitted code !!

### ISSUE THREAD

**Issue by D-R-L** [desktop/desktop#9140]
### Describe the bug
when there's an update tip on the top, I check branch and choose to bring my code changes to xxx branch, after that I found that all my code changes are lost !!! whatever I check between branches, there is still no code change found !!! omg I need help pls!!!!

### Version & OS
DESKTOP Version 2.3.1
macOS 10.14.6

**Comment by outofambit** [comment]
Hi @D-R-L, i'm sorry this happened! Fortunately, i think we can get your changes back.

1. Switch back to the branch you were working on when you lost your changes.

In a terminal in your repository:

2. run `git stash list`. you should see output like similar to

```shellsession
stash@{0}: WIP on detect-unready-fork-errors: b9d0424dc add error handler for fork unready errors
``` 

Note the part before the `:`, in this case it is `stash@{0}`. Take that and use is in the next command:

3. run `git stash apply stash@{0}`

You should see your changes (and likely some conflicts) in your working directory now. if you have any questions while you're doing this, please let us know! cc @desktop/support

**Comment by D-R-L** [comment]
> Hi @D-R-L, i'm sorry this happened! Fortunately, i think we can get your changes back.
> 
> 1. Switch back to the branch you were working on when you lost your changes.
> 
> In a terminal in your repository:
> 
> 1. run `git stash list`. you should see output like similar to
> 
> ```
> stash@{0}: WIP on detect-unready-fork-errors: b9d0424dc add error handler for fork unready errors
> ```
> 
> Note the part before the `:`, in this case it is `stash@{0}`. Take that and use is in the next command:
> 
> 1. run `git stash apply stash@{0}`
> 
> You should see your changes (and likely some conflicts) in your working directory now. if you have any questions while you're doing this, please let us know! cc @desktop/support

THX~~ I got my code back by using vsc's stash 😉 
But this desktop bug really shocked me… 😓
Maybe that update tip make this bug happened?

**Comment by billygriffin** [comment]
@D-R-L, apologies again and I'm so glad you were able to recover things. We were looking into this a bit more, and we suspect that it might be caused by the race condition that was fixed in #9038. That PR has been shipped to our beta version of GitHub Desktop, and I'd love if you'd be willing to try to reproduce this there to make sure you or others don't run into it again.

If you'd like to do so, you can download the beta here: https://github.com/desktop/desktop/#beta-channel

**Comment by D-R-L** [comment]
> @D-R-L, apologies again and I'm so glad you were able to recover things. We were looking into this a bit more, and we suspect that it might be caused by the race condition that was fixed in #9038. That PR has been shipped to our beta version of GitHub Desktop, and I'd love if you'd be willing to try to reproduce this there to make sure you or others don't run into it again.
> 
> If you'd like to do so, you can download the beta here: https://github.com/desktop/desktop/#beta-channel

Ok~ I'll try it 😃

**Comment by johannesschobel** [comment]
Dear @billygriffin or @D-R-L , i think, this issue can be closed! :+1:

**Comment by billygriffin** [comment]
Thanks @johannesschobel! ❤️ I'm going to close it as I think it was solved in #9038. If others are seeing this happen again, please let us know!

**Comment by nimmolo** [comment]
Happened to me just now. 

Had many uncommitted changes, working off my master repo. Made a new "feature" branch in GitHub Desktop, clicked "Bring my changes to new branch" instead of "Stash changes on current branch", changes disappeared. Many untracked files and renamed directories gone. 

Running git stash --list on the new "feature" branch shows nothing in the stash.

I'm not able to switch branches back to the "master" branch without resolving the 211 merge conflicts in the index of the new branch, and i'm scared to do that for fear of cementing the current file structure in place. 

I don't need the feature branch and would consider deleting it, if that would help. I'd really just like to check the stash of the "master" branch ... but i can't without switching.
